### PR TITLE
Fa

### PR DESCRIPTION
**Pull request by balanalina**

### CODE REVIEW

File: FA/fa/Finite_Automata.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+            elif command == 3:
+                for transition in self.transitions.keys():
+                    print(str(transition) + " --> " + str(self.transitions[transition]))
+            elif command == 4:
+                print(self.final_states)
+            elif command == 5:
+                print("Enter the sequence: ")
+                sequence = input()
+                if self.DFA_check(sequence):
+                    print(colored(sequence + " is accepted!",'green'))
```
Comment (CuckooHashing) [inline FA/fa/Finite_Automata.py]: I like the use of the colored library here, really makes the ui pop a bit more

File: FA/fa/Finite_Automata.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        self.transitions = {}
+        self.current_state = 'A'
+        self.read_file()
+
+    def read_file(self):
+        file = open(self.file, 'r')
+        fa = []
+        for line in file:
+            line.replace("\n","")
+            fa.append(line)
```
Comment (CuckooHashing) [inline FA/fa/Finite_Automata.py]: I don't quite understand the need for self.fa in this context, seems a bit redundant to me ....

File: FA/fa/Finite_Automata.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
+import sys
+from termcolor import colored, cprint
+
+class Finite_Automata:
```
Comment (CuckooHashing) [inline FA/fa/Finite_Automata.py:4]: This class seems to be tailor made for one specific example; could've been a bit more general, although that is easily solvable

File: FA/fa/Finite_Automata.py [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+        self.states = None
+        self.transitions = {}
+        self.current_state = 'A'
+        self.read_file()
+
+    def read_file(self):
+        file = open(self.file, 'r')
+        fa = []
+        for line in file:
+            line.replace("\n","")
```
Comment (CuckooHashing) [inline FA/fa/Finite_Automata.py]: A very elegant way to remove endlines, I personally like it.

File: FA/fa/Finite_Automata.py (hunk 2 above)
Comment (CuckooHashing) [reply FA/fa/Finite_Automata.py]: But it may be that I'm just bad a reading other people's code

**Review by CuckooHashing — COMMENTED** [review]
Wrote a review as requested in which I covered all that stuck out to me in terms of code/idea

File: FA/fa/Finite_Automata.py (hunk 3 above)
Comment (balanalina) [reply FA/fa/Finite_Automata.py:4]: It seems to but it should work with other ones too, I tried to make some changes to the fa.in and it worked accordingly  😄

File: FA/fa/Finite_Automata.py (hunk 2 above)
Comment (balanalina) [reply FA/fa/Finite_Automata.py]: You are right, I didn't use `self.fa` and forgot to erase it.
I use fa to have my file content in a list to resemble more the structure of the file as in `fa[1]` will be the second row from the file. I find it easier for later, e.g. i know that the alphabet is found on the second line so I will have it in `fa[1]`

**Pull request merged**